torch/testing/_internal/schema_check_mode.py

```diff
@@ -70,70 +85,86 @@ def unwrap(e):
                     return e.elem
                 except AttributeError as t:
                     return e
-            else:
-                return e
+            return e
+
+        def parse_metadata(e):
```

Comment:
Is the idea here that you want to assert that both the tensor data and metadata are unchanged after performing an op?

Are you able to use the `assertEqual()` method from the testing framework ([here](https://github.com/pytorch/pytorch/blob/09f79e94acce30e2002063322c9fe50d116361f1/torch/testing/_internal/common_utils.py#L2210)) instead? It should already handle checking metadata for you.

Another reason to use the one from the framework is that keeping track of the full set of metadata to check is tough, and can change over time. It looks like today, that method also checks `layout` and `device`: https://github.com/pytorch/pytorch/blob/09f79e94acce30e2002063322c9fe50d116361f1/torch/testing/_comparison.py#L635.

Reply:
From what I understand, the issue with assertEqual() is that torch.clone() doesn't copy the strides and storage cdata, so that assertEqual might return true when the strides have in fact changed or vice versa. Thus the parse_metadata function is there specifically to record data that isn't copied correctly in torch.clone().

Reply:
I agree with @goldenxuett , I think we need a separate check for these, because `clone()` doesn't copy strides or layout, and so we need to check those properties separately from the `assertEquals`